tensorflow/core/kernels/conv_ops_gpu_3.cu.cc

```diff
@@ -413,6 +429,275 @@ struct PadInput<GPUDevice, T, int, NDIMS> {
   }
 };
 
+// TPF stands for Tile size Posibility Frontier. It denotes the tile size
+// combinations that consume the most computational resources constrained by
+// number of threads per SM limit, shared memory limit and overall degree of
+// parallelism.
```

Comment:
Maybe this would be easier to read if we made it a list:

```
... constrained by the limits on
  - the number of threads per SM
  - shared memory per thread block, and
  - overall degree of parallelism.
```

except (and this is actually the important part) I don't really understand what the last one means.